src/rawtoaces_util/acesrender.cpp

```diff
@@ -106,275 +96,560 @@ void usage( const char *prog )
-//
-//	inputs:
-//      N/A
-//
-//	outputs:
-//      static AcesRender & : the referece to the only instance by calling
-//      the private getPrivateInstance() function
+    while ( pos < path.size() )
+    {
+        size_t end = path.find( separator, pos );
 
-AcesRender &AcesRender::getInstance()
+        if ( end == std::string::npos )
+            end = path.size();
+
+        std::string pathItem = path.substr( pos, end - pos );
+
+        if ( find( result.begin(), result.end(), pathItem ) == result.end() )
+            result.push_back( pathItem );
+
+        pos = end + 1;
+    }
+
+    return result;
+};
+
+bool fetch_camera_make_and_model(
+    const OIIO::ImageSpec &spec,
+    std::string           &camera_make,
+    std::string           &camera_model )
 {
+    camera_make = spec["cameraMake"];
+    if ( camera_make.empty() )
+    {
+        std::cerr << "Missing the camera manufacturer name in the file "
+                  << "metadata." << std::endl;
+        return false;
+    }
+
+    camera_model = spec["cameraModel"];
+    if ( camera_model.empty() )
+    {
+        std::cerr << "Missing the camera model name in the file metadata. "
+                  << std::endl;
+        return false;
+    }
 
-    return getPrivateInstance();
+    return true;
 }
 
-//	=====================================================================
-//	Initialize the single instance of the "AcesRender" class privately
-//
-//	inputs:
-//      N/A
-//
-//	outputs:
-//      static AcesRender &  : the referece to the only instance
-//      of "AcesRender" class
+vector<string> findFiles( string filePath, vector<string> searchPaths )
+{
+    vector<string> foundFiles;
+
+    for ( auto &i: searchPaths )
+    {
+        string path = i + "/" + filePath;
+
+        if ( std::filesystem::exists( path ) )
+            foundFiles.push_back( path );
+    }
+
+    return foundFiles;
+}
 
-AcesRender &AcesRender::getPrivateInstance()
+bool configure_solver(
+    core::Idt                      &solver,
+    const std::vector<std::string> &directories,
+    const std::string              &camera_make,
+    const std::string              &camera_model,
+    const std::string              &illuminant = "" )
 {
+    bool success = true;
+
+    auto camera_files = collect_data_files( directories, "camera" );
+    for ( auto &camera_file: camera_files )
+    {
+        success = solver.loadCameraSpst(
+            camera_file, camera_make.c_str(), camera_model.c_str() );
+        if ( success )
+            break;
+    }
+
+    if ( !success )
+    {
+        std::cerr << "Failed to find spectral data for camera " << camera_make
+                  << " " << camera_model << std::endl;
+        return false;
+    }
+
+    std::vector<std::string> found =
+        findFiles( "training/training_spectral.json", directories );
+    if ( found.size() )
+    {
+        // loading training data (190 patches)
+        solver.loadTrainingData( found[0] );
+    }
```

Comment:
```suggestion
    std::vector<std::string> found_training_data =
        findFiles( "training/training_spectral.json", directories );
    if ( found_training_data.size() )
    {
        // loading training data (190 patches)
        solver.loadTrainingData( found_training_data[0] );
    }
```
Also, do we care to let user know that there can be more then one and we pick the first one?